Sources/StackOrchestrator/StackOrchestrator+Store.swift

```diff
@@ -0,0 +1,278 @@
+import Foundation
+
+extension StackOrchestrator {
+
+    open class Store<NetworkStack, PersistenceStack>: StackOrchestratorStore
+    where
+        NetworkStack: Alicerce.NetworkStack,
+        PersistenceStack: Alicerce.PersistenceStack, PersistenceStack.Payload == NetworkStack.Remote
+    {
+
+        public typealias Payload = NetworkStack.Remote
+        public typealias Response = NetworkStack.Response
+
+        public typealias PersistenceKey = PersistenceStack.Key
+
+        public typealias Resource = FetchResource<NetworkStack.Resource, PersistenceStack.Key>
+
+        public typealias CompletionClosure<T, Response, E: Swift.Error> = (Result<FetchValue<T, Response>, E>) -> Void
+        public typealias FetchCompletionClosure<T> = CompletionClosure<T, Response, FetchError>
+
+        // MARK: - Properties
+
+        public let networkStack: NetworkStack
+        public let persistenceStack: PersistenceStack
+        public let performanceMetrics: StackOrchestratorPerformanceMetricsTracker?
+
+        // MARK: - Initialization
+
+        public init(
+            networkStack: NetworkStack,
+            persistenceStack: PersistenceStack,
+            performanceMetrics: StackOrchestratorPerformanceMetricsTracker?
+        ) {
+
+            self.networkStack = networkStack
+            self.persistenceStack = persistenceStack
+            self.performanceMetrics = performanceMetrics
+        }
+
+        // MARK: - Public Methods
+
+        @discardableResult
+        public func fetch(
+            resource: Resource,
+            completion: @escaping FetchCompletionClosure<Payload>
+        ) -> Cancelable {
+
+            // TODO: change the callback structure to allow returning multiple values (+ completion)
+
+            switch resource.strategy {
+            case .networkThenPersistence:
```

Comment:
Out of curiosity, and curiosity only, but can you give me a use case where persisted data is preferred over network data? Or am I misunderstanding the purpose of this?

Reply:
I think I understood this. If network fails we fetch the persisted data.

Reply:
That's precisely it 😄